File: mobile_app/lib/features/profile/models/profile_model.dart
class ProfileModel {
  final String id;
  final String name;
  final String avatarUrl;
  final bool isKids;
  final String? pin;
  final String languagePreference;

  ProfileModel({
    required this.id,
    required this.name,
    required this.avatarUrl,
    required this.isKids,
    this.pin,
    this.languagePreference = 'en',
  });

  factory ProfileModel.fromJson(Map<String, dynamic> json) {
    return ProfileModel(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      avatarUrl: json['avatarUrl'] ?? 'https://api.dicebear.com/7.x/bottts/png?seed=Vanix',
      isKids: json['isKids'] ?? false,
      pin: json['pin'],
      languagePreference: json['languagePreference'] ?? 'en',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      '_id': id,
      'name': name,
      'avatarUrl': avatarUrl,
      'isKids': isKids,
      'pin': pin,
      'languagePreference': languagePreference,
    };
  }
}

File: backend/src/models/profile.model.js
const mongoose = require('mongoose');

const ProfileSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  name: {
    type: String,
    required: [true, 'Please add a profile name'],
    trim: true,
    maxlength: [15, 'Profile name cannot exceed 15 characters'],
  },
  avatarUrl: {
    type: String,
    default: 'https://api.dicebear.com/7.x/bottts/png?seed=Vanix', // Premium visual avatar generator API
  },
  isKids: {
    type: Boolean,
    default: false,
  },
  pin: {
    type: String,
    default: null, // If set, profile requires this 4-digit pin to access
    match: [/^\d{4}$/, 'PIN must be a 4-digit number'],
  },
  watchlist: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Movie', // References movies or series
  }],
  languagePreference: {
    type: String,
    enum: ['en', 'te', 'ta', 'hi'],
    default: 'en',
  },
}, {
  timestamps: true,
});

module.exports = mongoose.model('Profile', ProfileSchema);
